casual/casual-jca/src/test/groovy/se/laz/casual/jca/pool/NetworkConnectionPoolTest.groovy

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+package se.laz.casual.jca.pool
+
+import se.laz.casual.internal.network.NetworkConnection
+import se.laz.casual.jca.Address
+import se.laz.casual.jca.CasualResourceAdapterException
+import se.laz.casual.network.ProtocolVersion
+import se.laz.casual.network.outbound.NetworkListener
+import spock.lang.Specification
+
+class NetworkConnectionPoolTest extends Specification
+{
+   def 'using the wrong address'()
+   {
+      given:
+      int poolSize = 5
+      def poolName = 'small-pool'
+      Address address = Address.of("nifty", 7771)
+      Address anotherAddress = Address.of('delta', 8787)
+      NetworkConnectionCreator connectionCreator = Mock(NetworkConnectionCreator){
+         1 * createNetworkConnection(*_) >> Mock(ReferenceCountedNetworkConnection)
```

Comment:
Please explain how it could mess up and returning a cached connection instance on first request due to this?
The connection creator is used inside the actual pool, in case a new ref counted connection should be created - that's what we're checking for, nothing else.

Reply:
My point is that wildcards with mocks are not good for testing - as you aren't actually testing the dependency conditions effectively.
Here you don't validate that the connection request is recieving the correct address information.
The implementation could be changed to be something bananas - like the cache example i provided - which though insane, would still pass you test.
If however you validate the parameters of the dependent call - the test would fail when the connection is performed with the wrong address for example, causing a regression if someone breaks the implementation in the future.

Reply:
I agree that we can check the address and also the protocol version, no worries there.
The rest of the params would be troublesome, to say the least, to verify - those would also not add anything to what is being tested.